Structure update

YuriiShp opened the pull request:
Now it is in usable condition

YuriiShp committed:
fixes in structure

YuriiShp committed:
fixes in structure

YuriiShp committed:
add README.md

YuriiShp committed:
correct README.md

YuriiShp committed:
move code_assessment.sh to the root

DzigMS commented on oct/pages/base/base.py:
```diff
@@ -58,10 +71,12 @@ class Page(Base):
 
     def __init__(self, refference: WebDriver = driver):
         super().__init__(refference)
+        print(self._reff)
```
WTF ????

DzigMS commented on oct/pages/base/base.py:
```diff
@@ -58,10 +71,12 @@ class Page(Base):
     url = ''
 
     @property
+    @timeout(5)
```
5 - magic number ???

YuriiShp committed:
add page.py module with base page

zoiaivanova commented on oct/pages/base/base.py:
```diff
@@ -37,6 +66,20 @@ def _setup(self):
+            element_class = self._check_class(self.loads.get('class'))
+            find_by = getattr(By, locator_name.upper())
+
+            elements = self._reff.find_elements(find_by, locator_value)
+            for index, element in enumerate(elements):
+                attr = ("%s_%d" % (element_class.__name__.lower(), index))
+
+                setattr(self, attr, element_class(element))
```
```suggestion
            elements = self._reff.find_elements(find_by, locator_value)
            attr = f"{element_class.__name__.lower()}s"
            setattr(self, attr, [element_class(element) for element in elements])
```

YuriiShp replied:
Can not apply list comprehension in that way since "attr" is inside the loop and changes its value on each iteration

zoiaivanova replied:
I suggested using one property for the list instead of a property per each item in a list.
instead of "product_thumb_1" and "product_thumb_2" it will be "product_thumbs" that returns a list of product thumb instances 
'attr' isn't in for loop in my suggestion

zoiaivanova commented on oct/pages/base/base.py:
```diff
@@ -15,16 +31,29 @@ def __init__(self, refference):
         },
     }
     """
+    loads = dict()
```
If it's needed for loading the list of elements like product thumb the name of a variable can be 'contains_lists' or 'list_items' instead of 'loads'. Or it can be in contains:
```
contains = {
        elementName: {
            "locator": (by, value),
            "class": Element
        },
       "listElement": {
           "locator": (by, value),
           "class": ListElement,
           "is_list": True
        }
    }
```

YuriiShp replied:
Ok, this is reasonable, going to apply changes

zoiaivanova commented on oct/pages/base/base.py:
```diff
@@ -1,10 +1,26 @@
+        def wrapper(self):
+            countdown = time.time()
+            while (time.time() - countdown) < timesec:
+                if func(self) is True:
+                    return func(self)
```
```suggestion
                func_result = func(self)
                if func_result is True:
                    return func_result
```

zoiaivanova commented on oct/pages/base/page.py:
```diff
@@ -0,0 +1,253 @@
+        self.go.click()
+
+
+class Descktop(Clickable):
```
'Desktop'

zoiaivanova commented on oct/pages/base/base.py:
```diff
@@ -110,3 +155,7 @@ class Element(Base):
     def __init__(self, refference: WebElement):
```
'reference' and 'ref'

YuriiShp committed:
add changes to base.py module

- added possibility to fill the page object with loadable content
- replaced self._reff with self._base
- correect grammar

YuriiShp committed:
rename develop folder to models

moiseienko-dg commented on oct/pages/base/base.py:
```diff
@@ -28,14 +62,32 @@ def _setup(self):
-                        self._get_instance(
-                            by=find_by, value=locator_value, elcls=element_class)
-                        )
+                if val.get('is_loaded') is True:
```
is True можно убрать, т.к. get возвращает bool type value
